src/index.ts

```diff
@@ -179,35 +188,36 @@ export default function VitePluginGraphQLCodegen(options?: Options): Plugin {
           return;
         }
 
-        if (matchOnDocuments) {
-          try {
-            const isDocument = await isGraphQLDocument(
-              filePath,
-              codegenContext,
-            );
-            log('Document check successful in file watcher');
-
-            if (!isDocument && !matchOnSchemas) return;
-
-            log('File matched a graphql document');
-          } catch (error) {
-            // GraphQL Codegen handles logging useful errors
-            log('Document check failed in file watcher');
-          }
-        }
-
-        if (matchOnSchemas) {
-          try {
-            const isSchema = await isGraphQLSchema(filePath, codegenContext);
-            log('Schema check successful in file watcher');
-
-            if (!isSchema) return;
-
-            log('File matched a graphql schema');
-          } catch (error) {
-            // GraphQL Codegen handles logging useful errors
-            log('Schema check failed in file watcher');
-          }
+        if (matchOnDocuments || matchOnSchemas) {
```

Comment:
The old behaviour was that if both were false it wouldn't return early (and would therefore generate types) which is what I was trying to preserve, I'd have preferred to do an early return otherwise to avoid the extra nesting. It could be that that was not the intended behaviour in which case I'll go ahead and make this change.